lib/score_logic.rb

```diff
@@ -0,0 +1,80 @@
+module Scrabble ##require your gems and classes here # require_relative 'scrabble/whatevs'
+
+  SCOREBOARD = {
+    "A" => 1, "E" => 1, "I" => 1, "O" => 1, "U" => 1,
+    "L" => 1, "N" => 1, "R" => 1, "S" => 1, "T" => 1,
+    "D" => 2, "G" => 2,
+    "B" => 3, "C" => 3, "M" => 3, "P" => 3,
+    "F" => 4, "H" => 4, "V" => 4, "W" => 4, "Y" => 4,
+    "K" => 5,
+    "J" => 8, "X" => 8,
+    "Q" => 10, "Z" => 10,
+  }
+
+  class Scrabble
+
+    def self.score(word)
+
+      return "Must pick a letter!" unless word[/[a-zA-Z]+/] == word && word.length > 0
+      #If the input is a character that isn't a letter, or if it's an empty string, it brings up an error
+
+      get_score_array(word)
+
+      sum = 0
+      # for each letter
+      @score_array.each do |score|
+        sum += score.to_i #adds all the letter scores together
+      end
+      #returns the sum/score of all letters
+      return sum
+    end
+
+    def self.get_score_array(word)
+      word = word.chars
+      # empty array to iterate through each letter of the split word
+      @score_array = []
+      #SCOREBOARD[letter.upcase] gives me the score value from my SCOREBOARD hash
+      word.each do |letter|
+        @score_array.push(SCOREBOARD[letter.upcase])
+      end
+      return @score_array
```

Comment:
Since you are returning the `@score_array` from this method, it would make more sense to just use a local variable in this method and then set that above on line 21 rather than using an instance variable.